p2p/proto/transaction.proto

```diff
@@ -61,15 +61,16 @@ message Transaction
         Felt252  address_salt     = 2;
         repeated Felt252 calldata = 3;
         uint32   version          = 4;
+        Address  address          = 5;
```

Comment:
I don't think we should be calculating hashes for the Deploy transaction. Juno decided that for `mainnet` and `goerli` we will begin syncing from a snapshot to avoid issues with backward compatibility. While `sepolia` doesn't have any deploy transactions.

Also, I realised that `version` should not have been added, rather the message should be renamed to `DeployV0`.

Reply:
Imo all transactions should also carry their respective hashes to allow discarding invalid messages as soon as a mismatch occurs.
There are 2 advantages here:
- we can identify a mismatch as soon as possible in case of "stupid" malicious nodes which just fill the message with garbage and blacklist them hopefully even without ingesting the entire response stream,
- in case of "clever" malicious nodes they need to spend computational resources to forge fake transactions that match their (declared) hashes which does make the attack a bit more expensive, still such response stream would ultimately be rejected because of transaction commitment mismatch.

Not having the hashes in those replies makes the "silly/garbage" attack more efficient because it always forces us to spend more computational resources before the node is blacklisted (we always calculate all the transaction hashes from the stream to then find out that the transaction commitment does not match).

Reply:
I think your reasoning can also be applied to including a block hash in the block header

Reply:
Yeah, I had the block header and class defs in mind as well.

Reply:
IIRC `version` needs to stay because some old deploys were marked as version 0 while others as version 1 and version goes into transaction hash calculation. (I'm not 100% sure though.)